source/MRViewer/ImGuiMenu.cpp

```diff
@@ -1196,8 +1196,7 @@ void ImGuiMenu::draw_object_recurse_( Object& object, const std::vector<std::sha
 
         }
 
-
-        if ( isSelected )
+        if ( isSelected && !getViewerInstance().mouseController.isPressed( MouseButton::Middle ) )
             drawSceneContextMenu_( selected );
```

Comment:
better inside this func call CloseCurrentPopup if middle is pressed (ImGuI::IsMouseDown[2])